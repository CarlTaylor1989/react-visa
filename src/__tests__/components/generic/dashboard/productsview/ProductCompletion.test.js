/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  ProductCompletion,
  mapStateToProps
} from '../../../../../components/generic/dashboard/productsview/ProductCompletion';
import {
  CHALLENGE_COMPLETION_STATUS,
  CHALLENGE_PERFECT_STATUS
} from '../../../../../config/constants';

let mockProps;
beforeEach(() => {
  mockProps = {
    challenges: [{
      id: 'c1',
      familyId: 'm1f2',
      productId: 'p3',
      status: CHALLENGE_COMPLETION_STATUS
    }, {
      id: 'c2',
      familyId: 'm1f2',
      productId: 'p3',
      status: CHALLENGE_PERFECT_STATUS
    }, {
      id: 'c3',
      familyId: 'm1f2',
      productId: 'p4',
      status: CHALLENGE_PERFECT_STATUS
    }],
    familyId: 'm1f2',
    productId: 'p3',
    closePopup: jest.fn()
  };
});

test('renders the ProductCompletion component', () => {
  const wrapper = shallow(<ProductCompletion {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('should mapStateToProps', () => {
  const state = {
    mapData: {
      challenges: []
    }
  };
  const props = mapStateToProps(state);
  expect(props).toEqual({
    ...state.mapData
  });
});
