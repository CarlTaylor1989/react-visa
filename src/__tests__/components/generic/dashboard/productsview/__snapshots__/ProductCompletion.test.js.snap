// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProductCompletion component 1`] = `
<div
  className="productCompletion"
>
  <div
    className="progress"
  >
    2
    <Translate
      tag="span"
      value="generic.dashboard.productProgressDivider"
    />
    2
  </div>
  <ProgressBar
    colour="dark"
    percent={100}
  />
</div>
`;
